compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs

```diff
@@ -542,7 +542,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             let span = self.tcx.hir().body(body_id).value.span;
             let ok = self
                 .at(&self.misc(span), self.param_env)
-                .eq(DefineOpaqueTypes::No, interior, witness)
+                .eq(DefineOpaqueTypes::Yes, interior, witness)
```

Comment:
We can keep this yes, since we never expect this to be anything other than equating a ty variable to a concrete type.